agent/tcs/client/client_test.go

```diff
@@ -602,7 +602,10 @@ func TestMetricsDisabled(t *testing.T) {
 		published <- struct{}{}
 	}).Return(nil).MinTimes(1)
 
-	go cs.Serve()
+	go func() {
```

Comment:
This change part of the PR?

Reply:
Yes sorry I meant to comment on this before sending for review.

This test randomly failed on me once and I was trying to debug, but I couldn't figure anything out without knowing what happened to this `Serve()` function call.

Afterwards I couldnt reproduce. If it's just very slightly flakey, at least in the future we can see if this function failed so I thought we should keep it in, since it's harmless to check this in a unit test anyways.

Reply:
+1

Reply:
+1